From 039cd98452fcf585533455c28200d438cda8ed7a Mon Sep 17 00:00:00 2001
From: Mohammed Gamal <mgamal@redhat.com>
Date: Tue, 8 Nov 2022 16:20:17 +0100
Subject: [PATCH 06/14] redhat: hv_set_if_config: Workaround for gateway
 numbering in NetworkManager

RH-Author: Miroslav Rezanina <mrezanin@redhat.com>
RH-MergeRequest: 9: Synchronize RHEL 9 changes to RHEL 10
RH-Jira: RHEL-40107 RHEL-40679
RH-Acked-by: Ani Sinha <anisinha@redhat.com>
RH-Commit: [6/14] 984b946aea905b94672aead098e73368e6c65fc7 (mrezanin/centos-git-hyperv-daemons)

Unlike older sysconfig scripts, NetworkManager expects GATEWAYx=ipaddr for all values of x.
So the first gateway is GATEWAY0 instead of GATEWAY. Other values should remain unchanged.
Workaround this by replacing GATEWAY= with GATEWAY0=.

A proper fix however, would be to generate NetworkManager keyfiles instead of ifcfg files.
That can be done eitter by changing hypervkvpd code to do that or to let the script parse
ifcfg files and generate corresponding NetworkManager keyfiles

Signed-off-by: Mohammed Gamal <mgamal@redhat.com>

patch_name: hpvd-redhat-hv_set_if_config-Workaround-for-gateway-numbe.patch
present_in_specfile: true
location_in_specfile: 9
---
 hv_set_ifconfig.sh | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/hv_set_ifconfig.sh b/hv_set_ifconfig.sh
index 146829b..fe7fccf 100644
--- a/hv_set_ifconfig.sh
+++ b/hv_set_ifconfig.sh
@@ -54,6 +54,10 @@ echo "IPV6INIT=yes" >> $1
 echo "PEERDNS=yes" >> $1
 echo "ONBOOT=yes" >> $1
 
+#Unlike older sysconfig scripts, NetworkManager expects GATEWAYx=ipaddr for all values of x.
+#So the first gateway is GATEWAY0 instead of GATEWAY. Other values should remain unchanged.
+#Workaround this by replacing GATEWAY= with GATEWAY0=.
+sed -i "s/GATEWAY=/GATEWAY0=/" $1
 
 cp $1 /etc/sysconfig/network-scripts/
 
-- 
2.39.3

